plugin.go

```diff
@@ -0,0 +1,129 @@
+// Package plasmactldependencies implements a dependencies launchr plugin
+package plasmactldependencies
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/launchrctl/keyring"
+	"github.com/launchrctl/launchr"
+	"github.com/launchrctl/launchr/pkg/cli"
+	plasmactlbump "github.com/skilld-labs/plasmactl-bump"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	launchr.RegisterPlugin(&Plugin{})
+}
+
+// Plugin is launchr plugin providing dependencies search action.
+type Plugin struct {
+	k keyring.Keyring
+}
+
+// PluginInfo implements launchr.Plugin interface.
+func (p *Plugin) PluginInfo() launchr.PluginInfo {
+	return launchr.PluginInfo{
+		Weight: 20,
+	}
+}
+
+// OnAppInit implements launchr.Plugin interface.
+func (p *Plugin) OnAppInit(app launchr.App) error {
+	app.GetService(&p.k)
+	return nil
+}
+
+// CobraAddCommands implements launchr.CobraPlugin interface to provide bump functionality.
+func (p *Plugin) CobraAddCommands(rootCmd *cobra.Command) error {
+	var depCmd = &cobra.Command{
+		Use:   "dependencies",
+		Short: "Shows parent and child resources of resource",
+		Args:  cobra.MatchAll(cobra.ExactArgs(1), cobra.OnlyValidArgs),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// Don't show usage help on a runtime error.
+			cmd.SilenceUsage = true
+
+			showPaths, err := cmd.Flags().GetBool("mrn")
+			if err != nil {
+				return err
+			}
+
+			return dependencies(args[0], !showPaths)
+		},
+	}
+
+	depCmd.Flags().Bool("mrn", false, "Show MRN instead of paths")
+	depCmd.SetArgs([]string{"target"})
+
+	rootCmd.AddCommand(depCmd)
+	return nil
+}
+
+func isMachineResourceName(target string) bool {
+	list := strings.Split(target, "__")
+	return len(list) == 3
+}
+
+func convertTarget(target string) (string, error) {
+	// @todo take current path as prefix
+	r := plasmactlbump.BuildResourceFromPath(target, "")
+	if r == nil {
+		return "", fmt.Errorf("not valid resource %q", target)
+	}
+
+	return r.GetName(), nil
+}
+
+func convertToPath(mrn string) string {
+	parts := strings.Split(mrn, "__")
+	return fmt.Sprintf("%s/%s/roles/%s", parts[0], parts[1], parts[2])
+}
+
+func dependencies(target string, toPath bool) error {
+	searchMrn := target
+	if !isMachineResourceName(searchMrn) {
+		converted, err := convertTarget(target)
+		if err != nil {
+			return err
+		}
+
+		searchMrn = converted
+	}
+
+	inv, err := plasmactlbump.NewInventory("empty", ".", "")
+	if err != nil {
+		return err
+	}
+	requiredMap := inv.GetRequiredMap()
+	children, ok := requiredMap[searchMrn]
+
+	if ok {
+		cli.Println("- Child dependencies:")
+		for item := range children {
+			res := item
+			if toPath {
+				res = convertToPath(res)
+			}
+
+			cli.Println("%s", res)
+		}
+	}
+
+	dependenciesMap := inv.GetDependenciesMap()
+	parents, ok := dependenciesMap[searchMrn]
```

Comment:
Output should recursively list dependencies of dependencies; as well as required of required

- Observed:
```
➜ ./plasmactl dependencies interaction/applications/roles/repositories 
- Child dependencies:
interaction/services/roles/repositories_nexus
```
- Expected:

```
➜ ./plasmactl dependencies interaction/applications/roles/repositories 
- Child dependencies:
interaction/services/roles/repositories_nexus
interaction/softwares/roles/nexus
```

Reply:
for instance with `➜ ./plasmactl dependencies interaction/applications/roles/mail`
we could get 
```

- Child dependencies:
cognition/softwares/roles/druid_exporter
cognition/softwares/roles/nats_kafka
foundation/services/roles/cluster_cni
foundation/softwares/roles/cni
foundation/softwares/roles/couchbase_exporter
foundation/softwares/roles/flannel_cni
foundation/softwares/roles/ingress_ldap_auth
foundation/softwares/roles/kube_certs_renew
foundation/softwares/roles/kubeadm
foundation/softwares/roles/kubectl
foundation/softwares/roles/kubelet
foundation/softwares/roles/openldap
foundation/softwares/roles/openldap_exporter
foundation/softwares/roles/prometheus_alert_machine
foundation/softwares/roles/prometheus_kafka_adapter
interaction/applications/roles/academy
interaction/applications/roles/landing
interaction/services/roles/account_alpine
interaction/services/roles/kb_alpine
interaction/services/roles/mail_dkimpy
interaction/services/roles/mail_dovecot
interaction/services/roles/mail_dovecot_exporter
interaction/services/roles/mail_postfix
interaction/services/roles/mail_postfix_exporter
interaction/services/roles/mail_rainloop
interaction/softwares/roles/dkimpy
interaction/softwares/roles/dovecot
interaction/softwares/roles/dovecot_exporter
interaction/softwares/roles/dovecot_ldap
interaction/softwares/roles/dovecot_ldap_sieve
interaction/softwares/roles/errors_server
interaction/softwares/roles/gitlab_api_extension
interaction/softwares/roles/gitlab_logger
interaction/softwares/roles/go_mmproxy
interaction/softwares/roles/grafana
interaction/softwares/roles/grafana_alert_machine
interaction/softwares/roles/openssh
interaction/softwares/roles/postfix
interaction/softwares/roles/postfix_exporter
interaction/softwares/roles/postfix_ldap
interaction/softwares/roles/rainloop
interaction/softwares/roles/skilld_account_machine
platform/softwares/roles/alpine
platform/softwares/roles/c
platform/softwares/roles/fetcher
platform/softwares/roles/golang
platform/softwares/roles/java
platform/softwares/roles/maven
platform/softwares/roles/nginx
platform/softwares/roles/npm
platform/softwares/roles/php
```

Reply:
and with `➜ ./plasmactl dependencies interaction/applications/roles/mail` 
we could get 
```
- Parent dependencies:
interaction/applications/roles/mail
- Child dependencies:
foundation/services/roles/cluster_cni
foundation/softwares/roles/cni
foundation/softwares/roles/flannel_cni
foundation/softwares/roles/ingress_ldap_auth
foundation/softwares/roles/kube_certs_renew
foundation/softwares/roles/kubeadm
foundation/softwares/roles/kubectl
foundation/softwares/roles/kubelet
foundation/softwares/roles/openldap
foundation/softwares/roles/prometheus_kafka_adapter
interaction/applications/roles/academy
interaction/applications/roles/landing
interaction/services/roles/account_alpine
interaction/services/roles/kb_alpine
interaction/softwares/roles/dkimpy
interaction/softwares/roles/dovecot
interaction/softwares/roles/go_mmproxy
interaction/softwares/roles/openssh
interaction/softwares/roles/postfix
interaction/softwares/roles/postfix_ldap
platform/softwares/roles/alpine
platform/softwares/roles/c
platform/softwares/roles/fetcher
platform/softwares/roles/java
platform/softwares/roles/maven
platform/softwares/roles/nginx
platform/softwares/roles/npm
platform/softwares/roles/php
```

Reply:
I'm thinking maybe we should add an option like depth=n to limit recursion, if needed by user

And  by default list all

Reply:
done